### PR TITLE
feature/SEAR-2188-prometheus-stats-authenticated-client

### PR DESCRIPTION
**Pull request by nbkincaid**
**Issue Link**
https://spothero.atlassian.net/browse/SEAR-2188

**Description**
This pr begins to lay the groundwork for producing the dashboards to see metrics by client. 

These two dashboards will be updated to parameterize on the new "authenticated_client" param. 

https://grafana.internal.spothero.com/d/FH0y-uFMz/backend-dashboard?orgId=1
https://grafana.internal.spothero.com/d/gHRLlZhGz/http-endpoints?orgId=1

### CODE REVIEW

File: http/metrics_test.go [hunk 1]
```diff
@@ -285,3 +288,38 @@ func TestMetricsRoundTrip(t *testing.T) {
...
+		},
+		{
+			name:     "user authenticated",
+			client:   "spothero",
+			expected: "spothero",
+		},
+		{
+			name:     "partner authenticated - lyft",
+			client:   "lyft",
+			expected: "lyft",
```
Comment (theisenmark) [inline http/metrics_test.go:311]: We should probably rename this since it's in a public repository

**Review by theisenmark — APPROVED** [review]

File: http/metrics_test.go (hunk 1 above)
Comment (nbkincaid) [reply http/metrics_test.go]: good call

**Pull request merged**